src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -196,5 +197,44 @@ private static void AssertDataDescriptor(MemoryStream memoryStream, bool hasData
             Assert.Equal(hasDataDescriptor ? 8 : 0, fileBytes[6]);
             Assert.Equal(0, fileBytes[7]);
         }
+
+        [Theory]
+        // General purpose bit flag must get the appropriate bit set if a file comment or an entry name is unicode
+        [InlineData("ascii", "ascii!!!", "utf-8", "U��������")]
+        [InlineData("utf-8", "U��������", "ascii", "ascii!!!")]
+        [InlineData("ascii", "ascii!!!", "latin1", "L��������")]
+        [InlineData("latin1", "L��������", "ascii", "ascii!!!")]
+        [InlineData("utf-8", "U��������", "latin1", "L��������")]
+        [InlineData("latin1", "L��������", "utf-8", "U��������")]
+        public static void Create_ZipArchiveEntry_DifferentEncodingsFullNameAndComment(string en1, string s1, string en2, string s2)
```

Comment:
It's hard to figure out what the argument names stand for.

```suggestion
        public static void Create_ZipArchiveEntry_DifferentEncodingsFullNameAndComment(string entryEncoding, string entryName string commentEncoding, string comment)
```

Reply:
This comment has not been addressed?